python/private/toolchains_repo.bzl

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Create a repository to hold the toolchains.
+
+This follows guidance here:
+https://docs.bazel.build/versions/main/skylark/deploying.html#registering-toolchains
+
+The "complex computation" in our case is simply downloading large artifacts.
+This guidance tells us how to avoid that: we put the toolchain targets in the
+alias repository with only the toolchain attribute pointing into the
+platform-specific repositories.
+"""
+
+PLATFORMS = {
```

Comment:
The way this dict is used appears to lead to failures where if you specify python 3.8 as the target, which does not have an artifact for `aarch64-apple-darwin`, Bazel will error out. Specifying the desired version should only define what's available.

Reply:
Fixed in https://github.com/bazelbuild/rules_python/pull/618/commits/0e91655421bdc5fd16a50ba9e47733fbeb45502f.